metadata/metadata.go

```diff
@@ -47,18 +51,30 @@ func FromContext(ctx context.Context) (Metadata, bool) {
 		return nil, ok
 	}
 
-	// capitalise all values
-	newMD := make(map[string]string)
+	newMD := make(Metadata)
+
 	for k, v := range md {
-		newMD[strings.Title(k)] = v
+		key, err := base32.StdEncoding.DecodeString(k)
+		if err != nil {
+			panic(err)
```

Comment:
Should you panic?

Reply:
sorry, debug stuff that never can happen